### ISSUE TITLE
Error: no client authentication mechanism provided

### ISSUE THREAD

**Issue by MatthiasGrandl** [auth0/express-openid-connect#84]
After logging in I get `no client authentication mechanism provided`. I use `oidc-provider` with the following client:

```
{
      client_id: "foo",
      client_secret: "bar",
      token_endpoint_auth_method: "client_secret_post",
      redirect_uris: ["http://localhost:3000", "http://localhost:3000/callback"],
      grant_types: ["authorization_code", "refresh_token"],
      response_types: ["code"]
}
```
This is my auth config:
```
{
    clientID: "foo",
    clientSecret: "bar",
    authorizationParams: {
      response_type: 'code',
      response_mode: 'form_post',
      scope: 'openid email offline_access'
    }
}
```

I can manually `curl` to get the tokens like described [here](https://auth0.com/docs/api-auth/tutorials/authorization-code-grant#2-exchange-the-authorization-code-for-an-access-token)

I would appreciate any pointers.

**Comment by adamjmcgrath** [comment]
This looks like it's been resolved by https://github.com/panva/node-oidc-provider/issues/697#issuecomment-612436061 

Feel free to reopen if not

**Comment by MatthiasGrandl** [comment]
I solved this issue by writing my own client. Not sure why I am the only one who encounters this issue, perhaps some other middleware causes issues. It's ok if this remains closed for now.

**Comment by panva** [comment]
Fixed by #86
